xbmc/Application.cpp

```diff
@@ -2765,14 +2765,11 @@ bool CApplication::PlayFile(CFileItem item, const std::string& player, bool bRes
     if (item.IsVideo())
     {
       // check lock when trying to play video via home/json
-      if (CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow() == WINDOW_HOME)
-      {
-        const std::shared_ptr<CProfileManager> profileManager =
-            CServiceBroker::GetSettingsComponent()->GetProfileManager();
-        if (profileManager->GetCurrentProfile().videoLocked() &&
-            !g_passwordManager.IsMasterLockUnlocked(true))
-          return true; // refuse to play
-      }
+      if (CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow() == WINDOW_HOME &&
+          CServiceBroker::GetSettingsComponent()->GetProfileManager()->
+            GetCurrentProfile().videoLocked() &&
+          !g_passwordManager.IsMasterLockUnlocked(true))
+        return true; // refuse to play
```

Comment:
squashing things together...
@garbear and/or @fuzzard one question in principle:
would you drop this for readability or stay with it for low powered machines like raspi?
have not tested if there's a noticable difference

Reply:
In general, unless it's critical path code (e.g. hammered often in a loop), most programming choices don't make enough of a difference to matter. Focus on readability, understandability and code clarity.

Maybe use the following:

```c++

if (item.IsVideo())
{
  // check lock when trying to play video via home/json
  bool bPlay = true;

  if (CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow() != WINDOW_HOME)
    bPlay = false;

  if (!CServiceBroker::GetSettingsComponent()->GetProfileManager()->
        GetCurrentProfile().videoLocked())
    bPlay = false;

  if g_passwordManager.IsMasterLockUnlocked(true))
    bPlay = false;

  if (!bPlay)
    return true; // refuse to play

  CUtil::ClearSubtitles();
}
```
You can precede some or all boolean tests with a comment that explains the test, for better clarity.

Please check my boolean algebra. It's early and the coffee hasn't kicked in yet :)

Reply:
okay thanks ☕️
decided to stay with a single „if“ because no further action wanted if we‘re not on home